flexbox/src/main/java/com/google/android/flexbox/FlexContainer.java

```diff
@@ -159,4 +159,74 @@
      * returned list are not reflected to the original list.
      */
     List<FlexLine> getFlexLines();
+
+    /**
+     * Returns the length of decoration (such as deviders) of the flex item
+     *
+     * @param childAbsoluteIndex           the absolute index of the flex item within the flex
+     *                                     container
+     * @param childRelativeIndexInFlexLine the relative index of the flex item within the flex line
+     * @param flexItem                     the flex item from which the lenght of the decoration is
+     *                                     calculated
+     * @return the length of the decoration. Note that the length of the flex item itself is not
+     * included in the result.
+     */
+    int getDecorationLength(int childAbsoluteIndex,
+            int childRelativeIndexInFlexLine, FlexItem flexItem);
+
+    /**
+     * @return the top padding of the flex container.
+     */
+    int getPaddingTop();
+
+    /**
+     * @return the left padding of the flex container.
+     */
+    int getPaddingLeft();
+
+    /**
+     * @return the right padding of the flex container.
+     */
+    int getPaddingRight();
+
+    /**
+     * @return the bottom padding of the flex container.
+     */
+    int getPaddingBottom();
+
+    /**
+     * Returns the child measure spec for its width.
+     *
+     * @param widthSpec the measure spec for the width imposed by the parent
+     * @param padding the padding along the width for the parent
+     * @param childDimension the value of the child dimension
+     */
+    int getChildWidthMeasureSpec(int widthSpec, int padding, int childDimension);
+
+    /**
+     * Returns the child measure spec for its height.
+     *
+     * @param heightSpec the measure spec for the height imposed by the parent
+     * @param padding the padding along the height for the parent
+     * @param childDimension the value of the child dimension
+     */
+    int getChildHeightMeasureSpec(int heightSpec, int padding, int childDimension);
+
+    /**
+     * Callback when a new flex item is added to the current container
+     *
+     * @param childAbsoluteIndex           the absolute index of the flex item added
+     * @param childRelativeIndexInFlexLine the relative index of the flex item added within the
+     *                                     flex line
+     * @param flexLine                     the flex line where the new flex item is added
+     */
+    void onNewFlexItemAdded(int childAbsoluteIndex, int childRelativeIndexInFlexLine,
```

Comment:
Not sure if we should differentiate these with "absolute" and "relative". Can these just be `index` and `indexInFlexLine`?